File: 3519-find-the-number-of-winning-players/find-the-number-of-winning-players.py
class Solution:
    def winningPlayerCount(self, n: int, pick: List[List[int]]) -> int:
        count=0
        hashmap={}
        for player,color in pick:
            if player not in hashmap:
                hashmap[player]=[color]
            else:
                hashmap[player].append(color)
        print(hashmap)
        for i in hashmap:
            hashmap2={}
            for color in hashmap[i]:
                if color in hashmap2:
                    hashmap2[color]+=1
                else:
                    hashmap2[color]=1
            freq=float("-inf")
            for val in hashmap2.values():
                if val>freq:
                    freq=val
            print(freq)
            if freq>i:
                count+=1
        return count

        

File: 3519-find-the-number-of-winning-players/find-the-number-of-winning-players.cpp
class Solution {
public:
    int winningPlayerCount(int n, vector<vector<int>>& pick) {
        unordered_map<int,unordered_map<int,int>> hash; // player-> [ball,count] 
        // cutu mwahh 
        // love you
        for(auto data: pick){
            int player= data[0], ballColor=data[1];
            hash[player][ballColor]++;
        }
        int winnersCount=0;
        for(auto data: hash){
            int player=data.first;
            bool flag=false;
            auto smallHash= data.second;
            for(auto pair: smallHash){
                int color=pair.first, count=pair.second;
                if(count>player){
                    flag=true;
                    break;
                }
            }
            if(flag)winnersCount++;    
        }
        return winnersCount;
    }
};